sdk/inc/azure/iot/az_iot_provisioning_client.h

```diff
@@ -322,6 +341,10 @@ typedef struct
  * @param[in] received_payload An #az_span containing the received MQTT payload.
  * @param[out] out_response If the message is register-operation related, this will contain the
  * #az_iot_provisioning_client_register_response.
+ * @pre \p client must not be `NULL`.
+ * @pre \p received_topic must be a valid span of size greater than or equal to 0.
+ * @pre \p received_payload must be a valid span of size greater than or equal to 0.
```

Comment:
note on these two. These are currently written to the spec of the API here:
https://github.com/Azure/azure-sdk-for-c/blob/66cd00042a12466d998811c9c504615f0d5afd29/sdk/src/azure/iot/az_iot_provisioning_client.c#L521-L522
I don't think we have any other cases where we have received topics or really spans for that matter where 0 is accepted as valid. Are we able to change these to make them more strict?
@ahsonkhan @RickWinter

Reply:
I remember @JeffreyRichter saying we can only relax req's to not break people.

Reply:
why would the topic be zero?

Reply:
that's what I'm saying. So I'd like to change these to minimum size 1 like the rest of the APIs but wanted to check with the SDK team if it's allowed

Reply:
In practice, is the minimum size of a valid topic actually 1? What's an example of such a topic string?

I think it's reasonable to consider strengthening the precondition if we are confident no customer would reasonably be impacted by such a change because no one would want to pass an empty topic. We can be pragmatic about what contractual strengthening/changes are actual breaks vs just bug/correctness fixes.

That said, let's do that outside of this precondition documentation PR.

Reply:
Looks like we need at least 23, if not more, looking at:
https://github.com/Azure/azure-sdk-for-c/blob/4b2f787bc155d1cd67f64ad1eebfdbc1438f6ce5/sdk/src/azure/iot/az_iot_provisioning_client.c#L21-L28

For example, this slice to end call would fail the precondition anyway, if `received_topic` is smaller than the expected size:
https://github.com/Azure/azure-sdk-for-c/blob/4b2f787bc155d1cd67f64ad1eebfdbc1438f6ce5/sdk/src/azure/iot/az_iot_provisioning_client.c#L536

https://github.com/Azure/azure-sdk-for-c/blob/222010d40d108e356896985277e14e91b5b9a5d8/sdk/src/azure/core/az_span.c#L60-L65

Reply:
gonna leave this open until @JeffreyRichter gives his 2 cents.